spec.bs

```diff
@@ -410,33 +410,61 @@ expiration time for the redemption record provided. This expiration is
 specified as number of seconds in the `Sec-Private-State-Token-Lifetime` HTTP response
 header value.
 
+<h3 id="the-document-object">Changes to {{Document}}</h3>
+
+<pre class="idl">
+partial interface Document {
+  Promise&lt;boolean> hasPrivateTokens(USVString issuer);
+  Promise&lt;boolean> hasRedemptionRecord(USVString issuer);
+};
+</pre>
 
 Query APIs {#query-apis}
 ========================
 
-Token Query {#token-query}
---------------------------
+To <dfn>determine whether associating an issuer would exceed the top-level limit</dfn> given an [=/origin=] |issuer| and an [=/origin=] |topLevel|, run the following steps:
 
-An origin might query whether the browser has any tokens from a specific issuer
-using the following API.
+1. Let |limit| be 2.
+2. TODO
 
-```javascript
-document.hasPrivateTokens(<issuer>)
-```
+To <dfn>associate the issuer</dfn> |issuer| (an [=/origin=]) with the [=/origin=] |topLevel|, run the following steps:
 
-Browser carries out the following steps on this query.
+1. TODO
 
- 1. Reject if the document is not in a secure context [[secure-contexts]].
- 2. Reject if associating the issuer with the top level origin would exceed the
-           top level’s number-of-issuers limit of 2.
- 3. Associate the issuer with the top level origin.
- 4. Look up the key commitments from storage. If there are key commitments,
-         discard tokens from issuer that are signed with keys other than those
-         from the issuer's most recent commitments.
- 5. Return true if there are tokens stored for the given issuer, false
-           otherwise.
+To <dfn>look up the key commitments</dfn> for a given [=/origin=] |issuer| and an [=/origin=] |topLevel|, run the following steps:
+
+1. TODO
+
+To determine whether an [=/origin=] |issuer| <dfn>is associated with</dfn> a given [=/origin=] |topLevel|, run the following steps:
 
-Note that this query modifies the browser state. It associates the issuer
+1. TODO
+
+To <dfn>discard tokens</dfn> from an [=/origin=] |issuer|, run the following steps:
+
+1. TODO
+
+Token Query {#token-query}
+--------------------------
+
+When invoked on {{Document}} |doc| with {{USVString}} |issuer|, the <dfn export method for=Document><code>hasPrivateTokens(issuer)</code></dfn> method must run these steps:
```

Comment:
I'm still not super familiar with the spec, but we have an issue argument that is sometimes an `origin` and sometimes a `USVString`... not sure if that's confusing or we should use `issuerOrigin` sometimes? For now, ignore this comment. Just thinking out loud.

Reply:
Just noting that all APIs have to take strings from web content and then parse them into origins (web content can't provide origins directly), so I think this is fine :)